src/engraving/layout/layoutsystem.cpp

```diff
@@ -431,6 +431,7 @@ System* LayoutSystem::collectSystem(const LayoutOptions& options, LayoutContext&
                 firstMeasure = false;
             }
             mb->setPos(pos);
+            mb->setParent(system);
```

Comment:
why would we need this? Does it mean that we're getting measures from system->measures() without any parent?

Reply:
This PR is not ready to merge yet and I've been spending the past few days trying to clean it up a little bit. That having been said, yes, measures don't have an explicitly set parent as they're being laid out (because we don't know what system the measure will be in until layout happens) but we need to set the parent as soon as possible once we know, because cross-staff stuff needs to know things like distance between staves in a system etc.

That's my understanding, at least, after talking to some people who did work on this before. Again this one is still a work in progress and should have something at least functional by the end of the week